### PR TITLE
python311Packages.asn1tools: disable failing tests

### PR DESCRIPTION
**Pull request by fabaff**
## Description of changes
Fix build (https://hydra.nixos.org/build/246592805)

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 281313` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>10 packages failed to build:</summary>
  <ul>
    <li>crackmapexec</li>
    <li>crackmapexec.dist</li>
    <li>netexec</li>
    <li>netexec.dist</li>
    <li>python312Packages.eth-account</li>
    <li>python312Packages.eth-account.dist</li>
    <li>python312Packages.eth-keyfile</li>
    <li>python312Packages.eth-keyfile.dist</li>
    <li>python312Packages.eth-keys</li>
    <li>python312Packages.eth-keys.dist</li>
  </ul>
</details>
<details>
  <summary>19 packages built:</summary>
  <ul>
    <li>asn1editor</li>
    <li>asn1editor.dist</li>
    <li>echidna</li>
    <li>python311Packages.aardwolf</li>
    <li>python311Packages.aardwolf.dist</li>
    <li>python311Packages.asn1tools</li>
    <li>python311Packages.asn1tools.dist</li>
    <li>python311Packages.eth-account</li>
    <li>python311Packages.eth-account.dist</li>
    <li>python311Packages.eth-keyfile</li>
    <li>python311Packages.eth-keyfile.dist</li>
    <li>python311Packages.eth-keys</li>
    <li>python311Packages.eth-keys.dist</li>
    <li>slither-analyzer (python311Packages.slither-analyzer)</li>
    <li>slither-analyzer.dist (python311Packages.slither-analyzer.dist)</li>
    <li>python311Packages.web3</li>
    <li>python311Packages.web3.dist</li>
    <li>python312Packages.asn1tools</li>
    <li>python312Packages.asn1tools.dist</li>
  </ul>
</details>

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 281313` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages failed to build:</summary>
  <ul>
    <li>crackmapexec</li>
    <li>crackmapexec.dist</li>
    <li>netexec</li>
    <li>netexec.dist</li>
  </ul>
</details>
<details>
  <summary>17 packages built:</summary>
  <ul>
    <li>asn1editor</li>
    <li>asn1editor.dist</li>
    <li>echidna</li>
    <li>python311Packages.aardwolf</li>
    <li>python311Packages.aardwolf.dist</li>
    <li>python311Packages.asn1tools</li>
    <li>python311Packages.asn1tools.dist</li>
    <li>python311Packages.eth-account</li>
    <li>python311Packages.eth-account.dist</li>
    <li>python311Packages.eth-keyfile</li>
    <li>python311Packages.eth-keyfile.dist</li>
    <li>python311Packages.eth-keys</li>
    <li>python311Packages.eth-keys.dist</li>
    <li>slither-analyzer (python311Packages.slither-analyzer)</li>
    <li>slither-analyzer.dist (python311Packages.slither-analyzer.dist)</li>
    <li>python311Packages.web3</li>
    <li>python311Packages.web3.dist</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
please cherry-pick https://github.com/NixOS/nixpkgs/commit/22554dc028889120fe2f072094068921424e8f63

File: pkgs/development/python-modules/asn1tools/default.nix [hunk 1]
```diff
@@ -42,14 +46,19 @@ buildPythonPackage rec {
     pytestCheckHook
   ] ++ lib.flatten (builtins.attrValues passthru.optional-depdendencies);
 
-
   pythonImportsCheck = [
     "asn1tools"
   ];
 
+  disabledTests = [
+    # https://github.com/eerimoq/asn1tools/issues/167
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/asn1tools/default.nix:54]:
```suggestion
    # assert exact error message of pyparsing which changed and no longer matches
    # https://github.com/eerimoq/asn1tools/issues/167
```

Comment (fabaff) [reply pkgs/development/python-modules/asn1tools/default.nix:57]: Thanks, added.

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 281313` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages failed to build:</summary>
  <ul>
    <li>crackmapexec</li>
    <li>crackmapexec.dist</li>
    <li>netexec</li>
    <li>netexec.dist</li>
  </ul>
</details>
<details>
  <summary>17 packages built:</summary>
  <ul>
    <li>asn1editor</li>
    <li>asn1editor.dist</li>
    <li>echidna</li>
    <li>python311Packages.aardwolf</li>
    <li>python311Packages.aardwolf.dist</li>
    <li>python311Packages.asn1tools</li>
    <li>python311Packages.asn1tools.dist</li>
    <li>python311Packages.eth-account</li>
    <li>python311Packages.eth-account.dist</li>
    <li>python311Packages.eth-keyfile</li>
    <li>python311Packages.eth-keyfile.dist</li>
    <li>python311Packages.eth-keys</li>
    <li>python311Packages.eth-keys.dist</li>
    <li>slither-analyzer (python311Packages.slither-analyzer)</li>
    <li>slither-analyzer.dist (python311Packages.slither-analyzer.dist)</li>
    <li>python311Packages.web3</li>
    <li>python311Packages.web3.dist</li>
  </ul>
</details>

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 281313` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>10 packages failed to build:</summary>
  <ul>
    <li>crackmapexec</li>
    <li>crackmapexec.dist</li>
    <li>netexec</li>
    <li>netexec.dist</li>
    <li>python312Packages.eth-account</li>
    <li>python312Packages.eth-account.dist</li>
    <li>python312Packages.eth-keyfile</li>
    <li>python312Packages.eth-keyfile.dist</li>
    <li>python312Packages.eth-keys</li>
    <li>python312Packages.eth-keys.dist</li>
  </ul>
</details>
<details>
  <summary>19 packages built:</summary>
  <ul>
    <li>asn1editor</li>
    <li>asn1editor.dist</li>
    <li>echidna</li>
    <li>python311Packages.aardwolf</li>
    <li>python311Packages.aardwolf.dist</li>
    <li>python311Packages.asn1tools</li>
    <li>python311Packages.asn1tools.dist</li>
    <li>python311Packages.eth-account</li>
    <li>python311Packages.eth-account.dist</li>
    <li>python311Packages.eth-keyfile</li>
    <li>python311Packages.eth-keyfile.dist</li>
    <li>python311Packages.eth-keys</li>
    <li>python311Packages.eth-keys.dist</li>
    <li>slither-analyzer (python311Packages.slither-analyzer)</li>
    <li>slither-analyzer.dist (python311Packages.slither-analyzer.dist)</li>
    <li>python311Packages.web3</li>
    <li>python311Packages.web3.dist</li>
    <li>python312Packages.asn1tools</li>
    <li>python312Packages.asn1tools.dist</li>
  </ul>
</details>

**Comment by hellwolf** [comment]
yes, please!

**Review by SuperSandro2000 — APPROVED** [review]

**Pull request merged**